src/panels/ActionsPanel/ActionsPanel.tsx

```diff
@@ -99,52 +167,117 @@ export default function ActionsPanel(): JSX.Element {
           Run action on {appName}: {generateSelectedUnitList()}
         </div>
         <div className="actions-panel__action-list">
-          {generateActionlist(
-            actionData,
-            fetchingActionData,
-            selectedAction,
-            setSelectedAction
-          )}
+          <LoadingHandler
+            data={data}
+            loading={fetchingActionData}
+            noDataMessage="This charm has not provided any actions."
+          >
+            {Object.keys(actionData).map((actionName) => (
+              <RadioInputBox
+                name={actionName}
+                description={actionData[actionName].description}
+                onSelect={selectHandler}
+                selectedInput={selectedAction}
+                key={actionName}
+              >
+                <ActionOptions
+                  name={actionName}
+                  data={actionData}
+                  onValuesChange={changeHandler}
+                />
+              </RadioInputBox>
+            ))}
+          </LoadingHandler>
+        </div>
+        <div className="actions-panel__drawer">
+          <button
```

Comment:
Not using the react-components button!? You're a maverick!

Reply:
Hah woops just habit.